lib/instrumentation/modules/express.js

```diff
@@ -12,12 +12,45 @@ module.exports = function (express, agent, version) {
     return express
   }
 
+  // express 5 moves the router methods onto a prototype
+  var routerProto = semver.satisfies(version, '^5')
+    ? (express.Router && express.Router.prototype)
+    : express.Router
+
+  var reportedSymbol = Symbol('reported')
+  var layerPatched = false
+  function patchLayer (layer) {
+    if (layerPatched) return
+    layerPatched = true
+
+    debug('shimming express.Router.Layer.handle_error function')
+
+    shimmer.wrap(layer.constructor.prototype, 'handle_error', function (orig) {
+      return function (err, req) {
+        if (!err[reportedSymbol]) {
+          err[reportedSymbol] = true
```

Comment:
I really like this approach. I can think of a few places in our code base where we should do that instead 😅